commit 8d0004cbd40305114bbe8019122319b43d25a772
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Tue Sep 8 12:34:39 2009 +0000

    Fix ATTRIBUTE_HIDDEN definition if __attribute__((visibility("hidden"))) is not supported.

commit 7a8bf489ffd436d853364dd0d3c9a6734b8cc7eb
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Thu Sep 3 15:46:37 2009 +0000

    Check for __attribute__((visibility("hidden"))).

diff --git a/configure.ac b/configure.ac
index da96872..abef435 100644
--- a/configure.ac
+++ b/configure.ac
@@ -132,6 +132,30 @@ if test x$enable_i965_driver = xyes && ! $PKG_CONFIG --atleast-version=2.4 libdr
 fi
 AM_CONDITIONAL(BUILD_I965_DRIVER, test x$enable_i965_driver = xyes)
 
+dnl Check for __attribute__((visibility()))
+AC_CACHE_CHECK([whether __attribute__((visibility())) is supported],
+    libva_cv_visibility_attribute,
+    [cat > conftest.c <<EOF
+int foo __attribute__ ((visibility ("hidden"))) = 1;
+int bar __attribute__ ((visibility ("protected"))) = 1;
+EOF
+    libva_cv_visibility_attribute=no
+    if ${CC-cc} -Werror -S conftest.c -o conftest.s >/dev/null 2>&1; then
+        if grep '\.hidden.*foo' conftest.s >/dev/null; then
+            if grep '\.protected.*bar' conftest.s >/dev/null; then
+	       libva_cv_visibility_attribute=yes
+            fi
+        fi
+    fi
+    rm -f conftest.[cs]
+])
+ATTRIBUTE_HIDDEN=""
+if test $libva_cv_visibility_attribute = yes; then
+    ATTRIBUTE_HIDDEN="__attribute__((visibility(\"hidden\")))"
+fi
+AC_DEFINE_UNQUOTED([ATTRIBUTE_HIDDEN], [$ATTRIBUTE_HIDDEN],
+    [Defined to __attribute__((visibility("hidden"))) when available])
+
 # We only need the headers, we don't link against the DRM libraries
 LIBVA_CFLAGS="$DRM_CFLAGS"
 AC_SUBST(LIBVA_CFLAGS)
